torch_geometric/transforms/rooted_subgraphs.py

```diff
@@ -0,0 +1,210 @@
+import torch
+from torch_cluster import random_walk
+from torch_sparse import SparseTensor
+
+from torch_geometric.data import Data
+from torch_geometric.transforms import BaseTransform
+
+
+class RootedSubgraphsData(Data):
+    r"""A data object describing a homogeneous graph together with each node's rooted subgraph.
+    It contains several additional properties that hold the information of all nodes' rooted subgraphs.
+    Assume the data represents a graph with :math:`N` nodes and :math:`M` edges, also each node
+    :math:`i\in \[N\]` has a rooted subgraph with :math:`N_i` nodes and :math:`M_i` edges.
+
+    Additional Properties:
+        subgraphs_nodes_mapper (LongTensor): map each node in rooted subgraphs to a node in the original graph.
```

Comment:
I'd like to rename these all to 'subgraph'. WDYT?

Reply:
Sounds good to me.